### ISSUE TITLE
Harmonic's fresh new build issues

### ISSUE THREAD

**Issue by jaydson** [JSRocksHQ/harmonic#184]
I'm trying to build Harmonic and it's not working properly.

Node.js version: 5.4.0
npm version: 3.3.12
OS: Ubuntu 14.04

Steps to reproduce:
- Clone harmonic's repo
- `npm link`
The `link` command seems to work.
But when I tried to run Harmonic on terminal, i have the following error:  
```
Error: Cannot find module '../dist/bin/cli/harmonic'
```

Then I tried to run `npm run dev`:  
```
[23:03:41] Watching src/ directory for changes...
/home/jaydson/Workspace/harmonic/node_modules/gulp-eslint/node_modules/eslint/lib/eslint.js:771
            location = node.loc.start;
                               ^

TypeError: Cannot read property 'start' of undefined
    at EventEmitter.module.exports.api.report (/home/jaydson/Workspace/harmonic/node_modules/gulp-eslint/node_modules/eslint/lib/eslint.js:771:32)
    at RuleContext.report (/home/jaydson/Workspace/harmonic/node_modules/gulp-eslint/node_modules/eslint/lib/rule-context.js:98:16)
    at /home/jaydson/Workspace/harmonic/node_modules/gulp-eslint/node_modules/eslint/lib/rules/no-undef.js:80:29
    at Array.forEach (native)
    at EventEmitter.Program:exit (/home/jaydson/Workspace/harmonic/node_modules/gulp-eslint/node_modules/eslint/lib/rules/no-undef.js:71:33)
    at emitOne (events.js:82:20)
    at EventEmitter.emit (events.js:169:7)
    at Controller.controller.traverse.leave (/home/jaydson/Workspace/harmonic/node_modules/gulp-eslint/node_modules/eslint/lib/eslint.js:732:25)
    at Controller.__execute (/home/jaydson/Workspace/harmonic/node_modules/gulp-eslint/node_modules/eslint/node_modules/estraverse/estraverse.js:397:31)
    at Controller.traverse (/home/jaydson/Workspace/harmonic/node_modules/gulp-eslint/node_modules/eslint/node_modules/estraverse/estraverse.js:485:28)

npm ERR! Linux 3.13.0-74-generic
npm ERR! argv "/home/jaydson/.nvm/versions/node/v5.4.0/bin/node" "/home/jaydson/.nvm/versions/node/v5.4.0/bin/npm" "run" "dev"
npm ERR! node v5.4.0
npm ERR! npm  v3.3.12
npm ERR! code ELIFECYCLE
npm ERR! harmonic@0.5.1 dev: `gulp`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the harmonic@0.5.1 dev script 'gulp'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the harmonic package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     gulp
npm ERR! You can get their info via:
npm ERR!     npm owner ls harmonic
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/jaydson/Workspace/harmonic/npm-debug.log
```

I'm investigating this right now, but I want to keep the history here in case others have the same issue.

**Comment by jaydson** [comment]
It looks like the documentation is wrong.
I have everything working now, here's the steps:

- Clone the repo
- `npm install`
- `npm run dev`
- `npm link`

Thoughts @UltCombo ?

**Comment by UltCombo** [comment]
I can confirm the build (`npm test` / `npm run dev`) is failing, seems to be due to a bug in ESLint or babel-eslint. We can try updating these to latest, though I'd like to land these changes on es20xx simultaneously. In fact, I already have the latest ESLint working on an es20xx-generated project, I just have to apply these changes to es20xx and Harmonic.

The `link` issue seems to be due to the failing build—as the build is failing, there is no `dist` directory (which is where all compiled code that is executed by Node resides).

**Comment by jaydson** [comment]
So, fixing the documentation and updating es20xx will solve the problem, right?

**Comment by UltCombo** [comment]
I'll update our ESLint-related dependencies and migrate our `.eslintrc` to the new version's format. That should fix the issue.

**Comment by jaydson** [comment]
:+1:

**Comment by UltCombo** [comment]
I'm updating npm and doing a clean `npm install` to make sure that the issue is fixed.

**Comment by UltCombo** [comment]
I've committed 77b6555c400b9fc93cdb34159c156366c5b90347, now `npm run dev` should at least generate the `dist` directory properly.

Though, I'm not getting a different error with ESLint:
> Cannot find module 'babel-runtime/helpers/interop-export-wildcard'
I'm not sure if this is a problem with my installation, but at least it is not breaking the build.

**Comment by UltCombo** [comment]
@jaydson Can you check if the build is working for you?

**Comment by jaydson** [comment]
You mean trying to build from zero again?

**Comment by UltCombo** [comment]
`git pull && rm -rf node_modules && npm i`

**Comment by jaydson** [comment]
It worked.

**Comment by UltCombo** [comment]
Is ESLint working correctly for you?

**Comment by UltCombo** [comment]
The problem with ESLint was that npm@3's maximally flat dependencies tree was using an outdated `babel-runtime` dependency for `babel-eslint` due to our own `babel-runtime` being outdated. This should fix it: 98f4486e8208bdda49c9c2c5a2ff36add3ff606c
